src/guides/v2.4/test/integration/annotations/magento-data-fixture-data-provider.md

```diff
@@ -0,0 +1,124 @@
+
+Inline JSON data provider is a JSON string with fixture aliases as keys.
+
+The following example shows how to use inline JSON as data provider:
+
+```php?start_inline=1
+class ProductsList extends \PHPUnit\Framework\TestCase
+{
+    /**
+     * @magentoDataFixture Magento\Catalog\Test\Fixture\Product as:product1
+     * @magentoDataFixture Magento\Catalog\Test\Fixture\Product as:product2
+     * @magentoDataFixture Magento\Catalog\Test\Fixture\Product as:product3
+     * @magentoDataFixtureDataProvider {"product1":{"sku":"simple1"}}
+     * @magentoDataFixtureDataProvider {"product2":{"sku":"simple2"}}
+     * @magentoDataFixtureDataProvider {"product2":{"sku":"simple3","status":2}}
+     */
+    public function testGetProductsCount(): void
+    {
+    }
+}
+```
+
+### A callable as data provider
+
+A callable data provider is any valid PHP callable that is publicly accessible. The following formats are supported:
+
+-  `methodName` a public method declared in the current test class
+-  `Namespace\Class::method` an external static method
+
+The following example shows how to use test class method as data provider:
+
+```php?start_inline=1
+class ProductsList extends \PHPUnit\Framework\TestCase
+{
+    /**
+     * @magentoDataFixture Magento\Catalog\Test\Fixture\Product as:product1
+     * @magentoDataFixture Magento\Catalog\Test\Fixture\Product as:product2
+     * @magentoDataFixture Magento\Catalog\Test\Fixture\Product as:product3
+     * @magentoDataFixtureDataProvider getProductsCountFixtureDataProvider
+     */
+    public function testGetProductsCount(): void
+    {
+    }
+
+    public function getProductsCountFixtureDataProvider(): array
+    {
+        return [
+            [
+                'product1' => [
+                    'sku' => 'simple1'
+                ],
+                'product2' => [
+                    'sku' => 'simple2'
+                ],
+                'product3' => [
+                    'sku' => 'simple3',
+                    'status' => Status::STATUS_DISABLED,
+                ],
+            ]
+        ];
+    }
+}
+```
+
+### Test case and test method scopes
+
+The `@magentoDataFixtureDataProvider` can be specified for a particular test or for an entire test case.
+The basic rules for Data Fixture Data Provider annotation at different levels are:
+
+-  `@magentoDataFixtureDataProvider` at a test case level, applies to all fixtures in the test case.
+-  `@magentoDataFixtureDataProvider` for a particular test, only applies to the fixtures declared at a test method level.
+
+### Restrictions
+
+`@magentoDataFixtureDataProvider` does not recursively merge fixtures data, thus if the fixture alias is defined in multiple `@magentoDataFixtureDataProvider`, only the last value will be used. `@magentoDataFixtureDataProvider` will not apply to a fixture that has an inline data provider with `with` directive.
+
+<!-- Link definitions -->
+
+[parameterizedDataFixture]: ../parameterized_data_fixture.html
```

Comment:
This link definition doesn't appear to be used anywhere. Please remove it.

Reply:
Done, thank you!